hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/cache/EntityCache.java

```diff
@@ -30,105 +30,102 @@
-                          .setEntityType(EntityType.SERVICE.name())
-                          .putIdentifyingAttributes(
-                              EntityConstants.getValue(CommonAttribute.COMMON_ATTRIBUTE_FQN),
-                              fqnAttribute)
-                          .build();
-
-                  return Optional.ofNullable(
-                      edsClient.getByTypeAndIdentifyingAttributes(pair.getLeft(), request));
-                }
-              });
-
+  private final LoadingCache<Pair<String, String>, Optional<Entity>> fqnToServiceEntityCache;
   /**
    * Cache to cache the service name to a list of services mapping so that we don't look it up over
    * and over.
    */
-  private final LoadingCache<Pair<String, String>, List<Entity>> nameToServiceEntities =
-      CacheBuilder.newBuilder()
-          .maximumSize(10000)
-          .expireAfterWrite(5, TimeUnit.MINUTES)
-          .build(
-              new CacheLoader<>() {
-                @Override
-                public List<Entity> load(@Nonnull Pair<String, String> pair) {
-                  // Lookup by name first, to see if there are any services with that name.
-                  return edsClient.getEntitiesByName(
-                      pair.getLeft(), EntityType.SERVICE.name(), pair.getRight());
-                }
-              });
+  private final LoadingCache<Pair<String, String>, List<Entity>> nameToServiceEntitiesCache;
 
   /**
    * Cache of K8S namespaces Key: Customer Id, Namespace name Value: List of Namespace entity ids
    */
-  private final LoadingCache<Pair<String, String>, List<Entity>> namespaceCache =
-      CacheBuilder.newBuilder()
-          .maximumSize(10000)
-          .expireAfterWrite(5, TimeUnit.MINUTES)
-          .build(
-              new CacheLoader<>() {
-                @Override
-                public List<Entity> load(@Nonnull Pair<String, String> key) {
-                  return edsClient.getEntitiesByName(
-                      key.getLeft(), EntityType.K8S_NAMESPACE.name(), key.getRight());
-                }
-              });
+  private final LoadingCache<Pair<String, String>, List<Entity>> nameToNamespaceEntitiesCache;
 
   /**
    * Cache of Backend identifying attributes to Entity Key: Map of identifying attributes Value:
    * Optional Backend entity
    */
   private final LoadingCache<ContextualKey<Map<String, AttributeValue>>, Optional<Entity>>
-      backendIdAttrsToEntityCache =
-          CacheBuilder.newBuilder()
-              .maximumSize(10000)
-              .expireAfterWrite(5, TimeUnit.MINUTES)
-              .build(CacheLoader.from(this::loadBackendFromIdentifyingAttributes));
+      backendIdAttrsToEntityCache;
 
-  public EntityCache(EdsClient edsClient) {
+  public EntityCache(EdsClient edsClient, Executor asyncCacheLoaderExecutor) {
     this.edsClient = edsClient;
+    fqnToServiceEntityCache =
+        CacheBuilder.newBuilder()
+            .maximumSize(10000)
```

Comment:
Ability to configure these caches is an existing problem. Will address them in a different PR.